addons/sale/models/account_invoice.py

```diff
@@ -52,7 +52,7 @@ def _onchange_partner_id(self):
         res = super(AccountMove, self)._onchange_partner_id()
 
         # Recompute 'narration' based on 'company.invoice_terms'.
-        if self.type == 'out_invoice':
+        if self.is_sale_document(include_receipts=True):
             self.narration = self.company_id.with_context(lang=self.partner_id.lang or self.env.lang).invoice_terms
```

Comment:
Shouldn't we move this code in account ?
Both narration & `company.invoice_terms` are specified in `account` module :thinking:.

cc: @smetl

Reply:
That's what I was thinking while investigating the bug but it's part of this big "_accounting-pocalypse_" commit : https://github.com/odoo/odoo/commit/bc131c0cfb51c953de8ec41fb820c8c7831eefb5#diff-a1623bbb015004f37de268fd661253ff4cc92da97bf99566e033b1c887a17dadR43 so there's a reason behind that I don't get for now :sweat_smile:

Reply:
It can be moved into `account` yes. I don't know why this is defined by `sale` at the first place

Reply:
Just moved it into `account`, it should be fine now :slightly_smiling_face: